### PR TITLE
feat: improve error handling with hint

### LINKED ISSUE #815: refine: More informative error message for `ValueError: target 'J' is not in this tree`
**Context**  
When I run `augur refine ... --root JN993010` but the root sequence is missing, I get the somewhat cryptic error:

```
Traceback (most recent call last):
  File "/usr/local/Caskroom/mambaforge/base/envs/nextstrain/bin/augur", line 10, in <module>
    sys.exit(main())
  File "/usr/local/Caskroom/mambaforge/base/envs/nextstrain/lib/python3.8/site-packages/augur/__main__.py", line 10, in main
    return augur.run( argv[1:] )
  File "/usr/local/Caskroom/mambaforge/base/envs/nextstrain/lib/python3.8/site-packages/augur/__init__.py", line 75, in run
    return args.__command__.run(args)
  File "/usr/local/Caskroom/mambaforge/base/envs/nextstrain/lib/python3.8/site-packages/augur/refine.py", line 245, in run
    T.root_with_outgroup(args.root)
  File "/usr/local/Caskroom/mambaforge/base/envs/nextstrain/lib/python3.8/site-packages/Bio/Phylo/BaseTree.py", line 844, in root_with_outgroup
    outgroup = self.common_ancestor(outgroup_targets, *more_targets)
  File "/usr/local/Caskroom/mambaforge/base/envs/nextstrain/lib/python3.8/site-packages/Bio/Phylo/BaseTree.py", line 447, in common_ancestor
    raise ValueError("target %r is not in this tree" % t)
ValueError: target 'J' is not in this tree
```

For some reason, only the first letter of the root is used `J` rather than `JN993010` which would be already better.

Best would be if the error from Bio Python was caught repackaged with a more useful message like `required root sequence named JN993010 not found in alignment/metadata`

### PR DESCRIPTION
**Pull request by corneliusroemer**
### Description of proposed changes    
Provides helpful error message when root is not found in tree

### Related issue(s)  
resolves #815

### CODE REVIEW

File: augur/refine.py [hunk 1]
```diff
@@ -242,7 +242,10 @@ def run(args):
             elif args.root in ['least-squares', 'min_dev', 'oldest']:
                 raise TypeError("The rooting option '%s' is only available when inferring a timetree. Please specify an explicit outgroup."%args.root)
             else:
-                T.root_with_outgroup(args.root)
+                try:
+                    T.root_with_outgroup(args.root)
+                except ValueError as err:
+                    raise ValueError(f"{err}\nHINT: This error may be because your specified root with name '{args.root}' was not found in your alignment file")
```
Comment (joverlee521) [inline augur/refine.py:248]: We can use exception chaining here to get the `The above exception was the direct cause of the following exception:` message in the tracebacks and not duplicate the `err` message.
```python3
except ValueError as err:
        raise ValueError("Custom error message...") from err
```

Comment (corneliusroemer) [reply augur/refine.py:248]: Sounds good!

Comment (corneliusroemer) [reply augur/refine.py]: Should be resolved now

**Review by joverlee521 — APPROVED** [review]

**Comment by emmahodcroft** [comment]
This seems like a good idea to me Cornelius! Helping to avoid cryptic error messages makes a huge difference. Thanks for tackling it!

**Review by emmahodcroft — APPROVED** [review]

**Review by victorlin — APPROVED** [review]

**Pull request merged**